APCAppCore/APCAppCore/UI/TabBarControllers/Activities/APCActivitiesViewController.m

```diff
@@ -493,16 +528,31 @@ - (void) fetchNewestSurveysAndTasksFromServer: (id) __unused sender
 - (void) updateWholeUI
 {
     [self.refreshControl endRefreshing];
-    [self performSelector:@selector(dismiss) withObject:self afterDelay:0.5];
+    [self performSelector:@selector(dismissSpinnerController) withObject:self afterDelay:0.5];
     [self configureNoTasksView];
     [self updateBadge];
     [self.tableView reloadData];
-    
 }
 
-- (void) dismiss
+- (void) dismissSpinnerController
 {
-    [self dismissViewControllerAnimated:YES completion:nil];
+    [self dismissViewControllerAnimated:YES completion:^{
+        if (shouldShowTodayActivitiesCompleteModal && self.countOfRemainingTasksToday == 0 && previousCountOfRemainingTasksToday == 1) {
+            [self showActivitiesCompleteModal];
```

Comment:
Not keen on this UX flow. Added a message cell to the daily activities section that is displayed once all the activities are completed is a nice UX addition. Adding more buttons that the user needs to push - not so much.